File: ReactNative/XkcdViewer/services/ApiService.ts
import IXkcdComic from '../data/IXkcdComic';

export class ApiService {
  public static async getComics({pageParam = 0}): Promise<IXkcdComic[]> {
    let currentComic = pageParam;
    let comics: IXkcdComic[] = [];
    for (let i = 0; i < 10; i++) {
      let response =
        currentComic === 0
          ? await fetch('https://xkcd.com/info.0.json')
          : await fetch(`https://xkcd.com/${currentComic}/info.0.json`);
      let comic = await response.json();
      comic.info = `${comic.year}-${comic.month.padStart(
        2,
        '0',
      )}-${comic.day.padStart(2, '0')} (${comic.num})`;
      comics.push(comic);
      currentComic = comic.num - 1;
    }
    return comics;
  }
}

File: Maui/XkcdViewer/Services/ApiService.cs
namespace XkcdViewer.Services;
public class ApiService
{
    private HttpClient _httpClient;

    public ApiService(IHttpClientFactory httpClientFactory)
    {
        _httpClient = httpClientFactory.CreateClient();    
    }

    public async Task<IEnumerable<XkcdComic>> GetComics(int currentComic = 0)
    {
        var result = new List<XkcdComic>();
        for(int i = 0; i < 10; i++)
        {
            var comic = await _httpClient.GetFromJsonAsync<XkcdComic>($"https://xkcd.com{(currentComic == 0 ? "" : $"/{currentComic}")}/info.0.json");
            currentComic = comic.num - 1;
            result.Add(comic);
        }
        return result;
    }
}
